lib/validator.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Validator do
+  @moduledoc """
+  The module that handles checking to see if the user input is numerical for a Tic-Tac-Toe game
+  """
+
+  @type message :: :non_numerical
+
+  @spec validate(String.t()) :: {atom(), non_neg_integer() | message}
+  def validate(user_input) when is_binary(user_input) do
```

Comment:
The `is_binary` guards against the `user_input` not being a string, this is actually more for the future validation checks (out of bounds), but since it doesn't do anything extra right now I thought it would be ok to include now. I can take it out for now if that would be better though! 😄

Reply:
ah right right! Sorry just jogging my Elixir memory, thanks Jon!

Reply:
No worries! Also, that is a really good point, I don't think so, I can add one now though! 😄

Reply:
I just added a test (and a small functionality change to the validator) to verify it guards against strings and if called with another form of a number (like an integer instead of a numerical string), it still works. Committing now!